src/frequenz/sdk/timeseries/battery_pool/_battery_pool.py

```diff
@@ -36,143 +33,132 @@
-        if self._batteries:
-            self._update_battery_status_task = asyncio.create_task(
-                self._update_battery_status(batteries_status_receiver)
-            )
-
-        self._min_update_interval = min_update_interval
-
-        self._power_distributing_sender = power_distributing_sender
-        self._active_methods: dict[str, MetricAggregator[Any]] = {}
+        self._battery_pool = battery_pool_ref
+        unique_id = str(uuid.uuid4())
+        self._source_id = unique_id if name is None else f"{name}-{unique_id}"
+        self._priority = priority
 
-        self._namespace: str = f"battery-pool-{self._batteries}-{uuid.uuid4()}"
-        self._power_distributing_namespace: str = f"power-distributor-{self._namespace}"
-        self._channel_registry: ChannelRegistry = channel_registry
-        self._formula_pool: FormulaEnginePool = FormulaEnginePool(
-            self._namespace,
-            channel_registry,
-            resampler_subscription_sender,
-        )
-
-    async def set_power(
+    async def propose_power(
         self,
-        power: Power,
+        power: Power | None,
         *,
-        adjust_power: bool = True,
         request_timeout: timedelta = timedelta(seconds=5.0),
         include_broken_batteries: bool = False,
+        bounds: timeseries.Bounds[Power | None] = timeseries.Bounds(None, None),
     ) -> None:
-        """Set the given power for the batteries in the pool.
+        """Send a proposal to the power manager for the pool's set of batteries.
 
         Power values need to follow the Passive Sign Convention (PSC). That is, positive
         values indicate charge power and negative values indicate discharge power.
 
-        When not using the Passive Sign Convention, the `charge` and `discharge` methods
-        might be more convenient.
+        If the same batteries are shared by multiple actors, the power manager will
+        consider the priority of the actors, the bounds they set, and their preferred
+        power, when calculating the target power for the batteries.
 
-        The result of the request can be accessed using the receiver returned from
-        the `power_distribution_results` method.
+        The preferred power of lower priority actors will take precedence as long as
+        they respect the bounds set by higher priority actors.  If lower priority actors
+        request power values outside of the bounds set by higher priority actors, the
+        target power will be the closest value to the preferred power that is within the
+        bounds.
+
+        When there are no other actors trying to use the same batteries, the actor's
+        preferred power would be set as the target power, as long as it falls within the
+        system power bounds for the batteries.
+
+        The result of the request can be accessed using the receiver returned from the
+        `power_status` method, which also streams the bounds that an actor should
```

Comment:
In general, I think it would be good to start including more cross-referencing in new code:
```suggestion
        [`power_status`][frequenz.sdk.timeseries.battery_pool.power_status] method, which also streams the bounds that an actor should
```

Reply:
Done here: 9eabfb59e7511adf599a704bfcf4646467c46527